modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/servlet/CacheControlFileEntryHttpHeaderCustomizer.java

```diff
@@ -79,6 +80,13 @@ private String _getHttpHeaderValue(FileEntry fileEntry, String currentValue)
 			return currentValue;
 		}
 
+		if (ArrayUtil.contains(
+				_cacheControlConfiguration.noCacheableMimeTypes(),
+				fileEntry.getMimeType())) {
+
+			return HttpHeaders.CACHE_CONTROL_NO_CACHE_VALUE;
+		}
+
```

Comment:
Shouldn't we move this block of code to precede the permission check? The "no-cache" policy is even more strict than the default policy (private); to me, it makes sense to let admins make things more restrictive.

Reply:
yes, you are right, we do not need to check permission before the no-cache policy.

I'll resend :)